### PR TITLE
Added weather information in the top right corner

### PR DESCRIPTION
**Pull request by C4illin**
The position of the weather information can be reworked, since it doesn't look that good where it is now.

I tried to follow the same code style, but my code still looks bad so any suggestion to rework that is appreciated.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -204,7 +205,76 @@ def calculate_minutes(departure):
     else:
         return f'Ca {countdown}'
 
-
+def get_temperature():
+    if temp_situation["updated"] == 0 or temp_situation["updated"] < datetime.now(tz.gettz("Europe/Stockholm")) - timedelta(minutes=10):
+        print("Updating temperatures")
+        openweathermap_key = os.environ["OWM_KEY"].strip()
```
Comment (edvinalestig) [inline app.py]: I thing you need a way to load the environment variable from a file. 
Something like this by the init stuff
`if "OWM_KEY" not in list(os.environ.keys()):
    with open("creds.txt") as f:
        owm_key = f.readlines()[-1]
    os.environ["OWM_KEY"] = owm_key`

File: app.py [hunk 2]
```diff
@@ -204,7 +205,76 @@ def calculate_minutes(departure):
...
+        hourly_r = requests.get(hourly_call)
+        hourly_json = json.loads(str(hourly_r.json()).replace("'", '"'))
+        temp0 = str(round((hourly_json["list"][0]["main"]["temp"]-272.15),1)) + "°C"
+        temp0time = datetime.fromtimestamp(hourly_json["list"][0]["dt"], tz.gettz("Europe/Stockholm")).strftime('%H:%M')
+        temp0wheather = getWeatherEmoji(hourly_json["list"][0]["weather"][0]["id"])
+        temp1 = str(round((hourly_json["list"][1]["main"]["temp"]-272.15),1)) + "°C"
+        temp1time = datetime.fromtimestamp(hourly_json["list"][1]["dt"], tz.gettz("Europe/Stockholm")).strftime('%H:%M')
+        temp1wheather = getWeatherEmoji(hourly_json["list"][1]["weather"][0]["id"])
+        temp2 = str(round((hourly_json["list"][2]["main"]["temp"]-272.15),1)) + "°C"
+        temp2time = datetime.fromtimestamp(hourly_json["list"][2]["dt"], tz.gettz("Europe/Stockholm")).strftime('%H:%M')
```
Comment (edvinalestig) [inline app.py]: Tips: Put `tz.gettz("Europe/Stockholm")).strftime('%H:%M')` in a variable because you use it a few times and it's not a pretty thing :)

**Review by edvinalestig — COMMENTED** [review]
It looks good! A good way to make your code look better is to have more whitespace. It makes the code much more readable. If you want to rework it a bit more you could try to move the functions to its own file and a class. You could make it more generic so you can use it in more projects if you wish. That way you don't have to rewrite stuff. It also makes it a bit more clear in my opinion.

The positioning of the info could be improved like you say. I don't know the best way but the logo should be moved so it's centered in the remaining whitespace. One idea I have is to maybe combine the weather and time/date div. Then you have the clock on top and the weather just below. Maybe?
Skärmbild (6)

Other than that, great job!

**Comment by C4illin** [comment]
Will move the code to a separate file when I get some more time. Should be done by Wednesday if I don't encounter any problems. Everything else you suggested is in the above commits. I like the idea of combining the temperature with the clock but I think it looks a bit weird with the Johanneberg logo of to one side. On the other hand that gives the weather information more well needed horizontal space, it looks a bit crammed as it is. Will have to think about it. 

Another idea would be to decrease the vertical space occupied by Chalmers and Kapellplatsen departures and put the weather information under them. But it doesn't look perfect either.
firefox_2019-11-17_22-59-02

Maybe moving down the arrivals and place the temperature on the same level as the clock div?
Here is a quick sketch of that:
firefox_2019-11-17_23-04-10


Thanks for all the help!

**Comment by edvinalestig** [comment]
I like how it looks when combining the divs. The thing is though that you can't shrink the departure div without the departures overflowing. As is is right now I think only one or two more rows fit. (5 tram lines, bus 16 with 3 destinations, 158 and 753 for a total of 17 rows in the rush hour.)

**Comment by C4illin** [comment]
I didn't think about that the number of lines varied so heavily, that makes my solution impossible. Is it possible to shrink the trafikstörningar div? Otherwise I think your solution is the best. Should we/I merge now and fix the layout later or wait until the layout is fixed first?

**Comment by C4illin** [comment]
What do you think about this layout?
firefox_2019-11-20_16-34-43
Unless there are more lines on kappelplatsen at some other time this should fit. But i don't like that the weather information is in the middle and destroys the symmetry between the departures divs. 

This was much harder than I first thought.

**Comment by edvinalestig** [comment]
Shrinking the disruptions div is a good idea. I tried it and got this: 
Screenshot_2019-11-20 Johanneberg
I think it looks rather good. Kapellplatsen shouldn't overflow normally, maybe if new lines are added because of reroutes or something. But I wouldn't worry about that.

**Comment by edvinalestig** [comment]
As a side note, in the future I think it's better if you create a branch in this repo instead of your fork (you have push access here) when you work on new things. That way it's easier to change between versions (at least for me).

**Comment by C4illin** [comment]
Thanks for the help with the layout!

I had my own fork since I started with this before I got push access but I should have moved over a long time ago. I will create a branch here in the future.

I think everything is ready for merge now.

**Pull request merged**